store.go

```diff
@@ -1162,14 +1164,18 @@ func (s *store) getROLayerStoresLocked() ([]roLayerStore, error) {
 		return nil, err
 	}
 
-	for _, store := range s.graphDriver.AdditionalImageStores() {
-		glpath := filepath.Join(store, driverPrefix+"layers")
+	additionalImageStores, additionalGraphDrivers := s.graphDriver.AdditionalImageStores()
 
-		rls, err := newROLayerStore(rlpath, glpath, s.graphDriver)
-		if err != nil {
-			return nil, err
+	for _, store := range additionalImageStores {
+		for _, driver := range additionalGraphDrivers {
+			glpath := filepath.Join(store, driver+"-layers")
+
+			rls, err := newROLayerStore(rlpath, glpath, s.graphDriver)
```

Comment:
Shouldn’t this use the right graph driver for that store?

Reply:
the only case we enabled now is overlay on top of vfs, and the overlay graph driver works well with layers stored as vfs.  I'll add a comment, not even sure how we would deal with multiple graphdrivers otherwise

Reply:
Right now, the overlay driver is, via `AdditionalImageStores()`, in charge; so I imagine it could
- detect the driver / format of the additional store
- fail on unexpected ones
- return the driver ID so that this line can use the right one.

See elsewhere.

I suppose I can live with the added comment…